cumulusci/tasks/salesforce/metadata_etl/sharing.py

```diff
@@ -0,0 +1,116 @@
+                    "internal_sharing_model and external_sharing_model is required."
+                )
+            self.owds[self._namespace_injector(elem["api_name"])] = {
+                "internal_sharing_model": elem.get("internal_sharing_model"),
+                "external_sharing_model": elem.get("external_sharing_model"),
+            }
+
+    def _post_deploy(self, result):
+        if result == "Success":
+            super()._post_deploy(result)
+            self.logger.info(f"Waiting for sharing enablement to complete.")
+            self.time_start = datetime.now()
+            self._poll()
+            self.logger.info(f"Sharing enablement is complete.")
+
+    def _transform_entity(self, metadata, api_name):
+        desired_internal_model = self.owds[api_name].get("internal_sharing_model")
+        desired_external_model = self.owds[api_name].get("external_sharing_model")
+
+        if desired_external_model:
+            external_model = metadata.findall(
+                f".//sf:externalSharingModel", self.namespaces
+            )
+            if not external_model:
+                external_model = [
+                    XML_ET.SubElement(
+                        metadata.getroot(),
+                        "{%s}externalSharingModel" % (self.namespaces.get("sf")),
+                    )
+                ]
+            external_model[0].text = desired_external_model
+
+        if desired_internal_model:
+            internal_model = metadata.findall(f".//sf:sharingModel", self.namespaces)
+            if not internal_model:
+                internal_model = [
+                    XML_ET.SubElement(
+                        metadata.getroot(),
+                        "{%s}sharingModel" % (self.namespaces.get("sf")),
+                    )
+                ]
+            internal_model[0].text = desired_internal_model
+
+        return metadata
+
+    def _poll_action(self):
+        elapsed = datetime.now() - self.time_start
+        if elapsed.total_seconds() > self.options["timeout"]:
+            raise CumulusCIException(
+                f'Sharing enablement not completed after {self.options["timeout"]} seconds'
+            )
+
+        for sobject in self.owds:
+            result = self.sf.query(
+                f"SELECT ExternalSharingModel, InternalSharingModel "
+                f"FROM EntityDefinition "
+                f"WHERE QualifiedApiName = '{sobject}'"
+            )
+            if result["totalSize"] == 1:
+                record = result["records"][0]
+                if (
+                    self.owds[sobject]["internal_sharing_model"]
+                    and record["InternalSharingModel"]
+                    != self.owds[sobject]["internal_sharing_model"]
+                ) or (
+                    self.owds[sobject]["external_sharing_model"]
+                    and record["ExternalSharingModel"]
+                    != self.owds[sobject]["external_sharing_model"]
+                ):
```

Comment:
Yeah. I just found this route slightly clearer. Open to argument.

Reply:
I thought about it a bit more and I think your approach is the right one. Otherwise you'd have to have some logic to toggle whether a field should be included in the filter.

Reply:
Oh, another observation would be that a NamedTuple in place of a dict is something to consider the next time around. 

Define the NamedTuple as:
```python
SharingModel = namedtuple("SharingModel", ["internal", "external"])
self.owds[self._namespace_injector(elem["api_name"])] = SharingModel(
    internal=elem.get("internal_sharing_model"),
    external=elem.get("external_sharing_model"),
)
```

This section would become:
```python
models = self.owds[sobject]
if result["totalSize"] == 1:
    record = result["records"][0]
    if (
        record["InternalSharingModel"] != models.internal
        or record["ExternalSharingModel"] != models.external
    ):
        return
```